File: frontend/types/system.ts
export interface SystemInfo {
  status: string;
  version: string;
  uptime?: number;
  last_update?: string;
}

export interface SystemMetrics {
  id: number;
  timestamp: string;
  cpu_usage?: number;
  cpu_temperature?: number;
  cpu_frequency?: number;
  memory_usage?: number;
  memory_available?: number;
  memory_total?: number;
  disk_usage?: number;
  disk_available?: number;
  disk_total?: number;
  network_in?: number;
  network_out?: number;
  system_status: string;
  uptime?: number;
}

export interface Process {
  id: number;
  pid: number;
  name: string;
  command?: string;
  cpu_usage?: number;
  memory_usage?: number;
  disk_usage?: number;
  status?: string;
  owner?: string;
  priority?: string;
  created_at: string;
  updated_at: string;
}

export interface Service {
  id: number;
  name: string;
  display_name?: string;
  description?: string;
  status: string;
  start_type: string;
  path?: string;
  created_at: string;
  updated_at: string;
}

export interface NetworkInterface {
  id: number;
  name: string;
  display_name?: string;
  mac_address?: string;
  ip_address?: string;
  netmask?: string;
  gateway?: string;
  status: string;
  speed?: number;
  created_at: string;
  updated_at: string;
}

export interface SecurityEvent {
  id: number;
  timestamp: string;
  event_type: string;
  severity: string;
  source?: string;
  description: string;
  user_id?: number;
  ip_address?: string;
  resolved: boolean;
  created_at: string;
}

export interface User {
  id: number;
  username: string;
  email: string;
  full_name?: string;
  role: string;
  is_active: boolean;
  is_superuser: boolean;
  last_login?: string;
  created_at: string;
  updated_at: string;
}

export interface Token {
  access_token: string;
  token_type: string;
  expires_in: number;
  user: User;
}

export interface SystemOverview {
  system_info: SystemInfo;
  current_metrics?: SystemMetrics;
  active_processes_count: number;
  running_services_count: number;
  network_interfaces_count: number;
  security_alerts_count: number;
}

export interface ProcessList {
  processes: Process[];
  total: number;
  page: number;
  size: number;
}

export interface ServiceList {
  services: Service[];
  total: number;
  page: number;
  size: number;
}

export interface NetworkInterfaceList {
  interfaces: NetworkInterface[];
  total: number;
  page: number;
  size: number;
}

export interface SecurityEventList {
  events: SecurityEvent[];
  total: number;
  page: number;
  size: number;
}

export interface AIInsight {
  id: string;
  type: 'optimization' | 'security' | 'performance' | 'maintenance';
  severity: 'low' | 'medium' | 'high' | 'critical';
  title: string;
  description: string;
  recommendation: string;
  confidence: number;
  timestamp: string;
  applied: boolean;
}

export interface Alert {
  id: string;
  type: 'info' | 'warning' | 'error' | 'success';
  title: string;
  message: string;
  timestamp: string;
  read: boolean;
  actions?: Array<{
    label: string;
    action: string;
  }>;
}

File: backend/api/schemas/system.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

# Base schemas
class DiskInfo(BaseModel):
    device: str
    mountpoint: str
    fstype: str
    total_size: float
    used_size: float
    free_size: float
    percent_used: float

class SystemInfo(BaseModel):
    # System
    hostname: str
    platform: str
    architecture: str
    os_version: str
    boot_time: str
    uptime: Optional[float] = None

    # CPU
    cpu_model: str
    cpu_cores_physical: int
    cpu_cores_logical: int

    # Memory
    total_memory: float
    used_memory: float
    available_memory: float
    
    # Swap
    total_swap: float
    used_swap: float
    free_swap: float

    # Disks
    disks: List[DiskInfo]

class SystemMetricsBase(BaseModel):
    cpu_usage: Optional[float] = None
    cpu_temperature: Optional[float] = None
    cpu_frequency: Optional[float] = None
    memory_usage: Optional[float] = None
    memory_available: Optional[float] = None
    memory_total: Optional[float] = None
    disk_usage: Optional[float] = None
    disk_available: Optional[float] = None
    disk_total: Optional[float] = None
    network_in: Optional[float] = None
    network_out: Optional[float] = None
    system_status: str = "operational"
    uptime: Optional[float] = None

class SystemMetricsCreate(SystemMetricsBase):
    pass

class SystemMetrics(SystemMetricsBase):
    id: int
    timestamp: datetime
    
    class Config:
        from_attributes = True

class ProcessBase(BaseModel):
    pid: int
    name: str
    command: Optional[str] = None
    cpu_percent: Optional[float] = None
    memory_percent: Optional[float] = None
    status: Optional[str] = None

class ProcessCreate(ProcessBase):
    pass

class Process(ProcessBase):
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

class ServiceBase(BaseModel):
    name: str
    display_name: Optional[str] = None
    description: Optional[str] = None
    status: str = "stopped"
    start_type: str = "manual"
    path: Optional[str] = None

class ServiceCreate(ServiceBase):
    pass

class Service(ServiceBase):
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

class NetworkInterfaceBase(BaseModel):
    name: str
    display_name: Optional[str] = None
    mac_address: Optional[str] = None
    ip_address: Optional[str] = None
    netmask: Optional[str] = None
    gateway: Optional[str] = None
    status: str = "down"
    speed: Optional[int] = None

class NetworkInterfaceCreate(NetworkInterfaceBase):
    pass

class NetworkInterface(NetworkInterfaceBase):
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

class SecurityEventBase(BaseModel):
    event_type: str
    severity: str
    source: Optional[str] = None
    description: str
    ip_address: Optional[str] = None
    resolved: bool = False

class SecurityEventCreate(SecurityEventBase):
    pass

class SecurityEvent(SecurityEventBase):
    id: int
    timestamp: datetime
    user_id: Optional[int] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

# Response schemas
class SystemOverview(BaseModel):
    system_info: SystemInfo
    current_metrics: Optional[SystemMetrics] = None
    active_processes_count: int
    running_services_count: int
    network_interfaces_count: int
    security_alerts_count: int

class ProcessList(BaseModel):
    processes: List[Process]
    total: int
    page: int
    size: int

class ServiceList(BaseModel):
    services: List[Service]
    total: int
    page: int
    size: int

class NetworkInterfaceList(BaseModel):
    interfaces: List[NetworkInterface]
    total: int
    page: int
    size: int

class SecurityEventList(BaseModel):
    events: List[SecurityEvent]
    total: int
    page: int
    size: int
